formats/cbor/commonMain/src/kotlinx/serialization/cbor/Cbor.kt

```diff
@@ -18,45 +15,111 @@ import kotlinx.serialization.modules.*
 ) : BinaryFormat {
 
     /**
-     * The default instance of [Cbor]
+     * The default instance of [Cbor]. Neither writes nor verifies tags. Uses indefinite length encoding by default.
      */
-    public companion object Default : Cbor(false, false, EmptySerializersModule())
+    public companion object Default :
+        Cbor(
+            CborConfiguration(
+                encodeDefaults = false,
+                ignoreUnknownKeys = false,
+                encodeKeyTags = false,
+                encodeValueTags = false,
+                encodeObjectTags = false,
+                verifyKeyTags = false,
+                verifyValueTags = false,
+                verifyObjectTags = false,
+                useDefiniteLengthEncoding = false,
+                preferCborLabelsOverNames = false,
+                alwaysUseByteString = false
+            ), EmptySerializersModule()
+        ) {
+
+        /**
+         * Preconfigured instance of [Cbor] for COSE compliance. Writes and verifies all tags, uses definite length
+         * encoding and prefers labels to serial names. **DOES NOT** sort CBOR map keys; declare them in canonical order
+         * for full cbor compliance!
+         */
+        public val CoseCompliant: Cbor =
+            Cbor {
+                encodeDefaults = false
+                ignoreUnknownKeys = false
+                encodeKeyTags = true
+                encodeValueTags = true
+                encodeObjectTags = true
+                verifyKeyTags = true
+                verifyValueTags = true
+                verifyObjectTags = true
+                useDefiniteLengthEncoding = true
+                preferCborLabelsOverNames = true
+                alwaysUseByteString = false
+                serializersModule = EmptySerializersModule()
+            }
+    }
 
     override fun <T> encodeToByteArray(serializer: SerializationStrategy<T>, value: T): ByteArray {
         val output = ByteArrayOutput()
-        val dumper = CborWriter(this, CborEncoder(output))
+        val dumper = if (configuration.useDefiniteLengthEncoding) DefiniteLengthCborWriter(
+            this,
+            output
+        ) else IndefiniteLengthCborWriter(
+            this,
+            output
+        )
         dumper.encodeSerializableValue(serializer, value)
+
         return output.toByteArray()
+
     }
 
     override fun <T> decodeFromByteArray(deserializer: DeserializationStrategy<T>, bytes: ByteArray): T {
         val stream = ByteArrayInput(bytes)
-        val reader = CborReader(this, CborDecoder(stream))
+        val reader = CborReader(this, CborParser(stream, configuration.verifyObjectTags))
         return reader.decodeSerializableValue(deserializer)
     }
 }
 
 @OptIn(ExperimentalSerializationApi::class)
-private class CborImpl(encodeDefaults: Boolean, ignoreUnknownKeys: Boolean, serializersModule: SerializersModule) :
-    Cbor(encodeDefaults, ignoreUnknownKeys, serializersModule)
+private class CborImpl(
```

Comment:
minor: I believe it can accept just configuration and redirect to `Cbor` constructor (like in `Json`)
Overall, IMO, it would be nice to have even just small implementation details be aligned between formats, so that those who will want to develop custom formats will see the pattern and the whole ecosystem of formats will be aligned on public API and internals somehow. But it is a minor and can be done in a follow-up PR if needed.

Reply:
I was working with what I got and did not really consider aligning across formats, but you are right, of course!

Reply:
done in 2294cc7719d0ea120ccd3970ed638b06f6383c9c